### PR TITLE
[PoC] Build ironic image from source

### PR DESCRIPTION
**Pull request by elfosardo**

### CODE REVIEW

File: main-packages-list.ocp [hunk 1]
```diff
@@ -12,47 +12,8 @@ ipxe-bootimgs-aarch64
...
-python3-debtcollector >= 2.5.0-0.20221128140303.a6b46c5.el9
-python3-dracclient >= 8.0.0-0.20221128135758.9c7499c.el
-python3-eventlet >= 0.33.1-4.el9
-python3-gunicorn >= 20.0.4-2.el9
-python3-ironic-lib >= 5.4.1-0.20230328130611.6e7574c.el9
-python3-ironic-prometheus-exporter >= 4.2.0-0.20230308192702.e63fb2e.el9
-python3-ironicclient >= 4.9.0-0.20211209154934.6f1be06.el9
 python3-jinja2 >= 2-3.0.1-2.el9.1
-python3-keystoneauth1 >= 5.0.0-0.20221128144522.2445a5d.el9
 python3-mod_wsgi
```
Comment (dtantsur) [inline main-packages-list.ocp:16]: Side note: I wonder if we still need this

**Comment by dtantsur** [comment]
Whatever we decide here, we should make sure it's aligned with upstream metal3 as much as possible. The requirements.txt thing is not there, right?

File: main-packages-list.ocp (hunk 1 above)
Comment (elfosardo) [reply main-packages-list.ocp:16]: fun fact: I removed the package at the beginning and httpd was complaining about that
now that I've added it back I'm getting a much worse error!
I'l try adding it back again and see if we can move forward from there

**Comment by elfosardo** [comment]
> Whatever we decide here, we should make sure it's aligned with upstream metal3 as much as possible. The requirements.txt thing is not there, right?

@dtantsur the plan is to propose switching to source code as main way to build the ironic image in metal3
it's slightly different from how it works here but the result is the same
here we're constrained by cachito (that's why the requirements.txt), while in metal3 we just install directly using normal requirements plus upper-constraints

**Comment by elfosardo** [comment]
the current failures apparently are widespread across other jobs and projects, not strictly related to this

File: main-packages-list.ocp (hunk 1 above)
Comment (elfosardo) [reply main-packages-list.ocp]: without that package we end up with httpd not starting, complaining with `httpd: Syntax error on line 61 of /etc/httpd/conf/httpd.conf: Syntax error on line 8 of /etc/httpd/conf.modules.d/httpd-modules.conf: Cannot load modules/mod_wsgi_python3.so into server: /etc/httpd/modules/mod_wsgi_python3.so: cannot open shared object file: No such file or directory
`

**Comment by elfosardo** [comment]
this works as intended
we need to define some things before putting it in production but it's a good start

**Comment by elfosardo** [comment]
the test went well but we've decidede to go for an hybrid model
this is superseeded by https://github.com/openshift/ironic-image/pull/444

**Pull request closed without merging**